### PR TITLE
CAPZ v26.0.0

### PR DESCRIPTION
**Pull request by njuettner**
Towards: https://github.com/giantswarm/roadmap/issues/3571

### Checklist
- [ ] Roadmap issue created
- [ ] Release uses latest stable Flatcar
- [ ] Release uses latest Kubernetes patch version

### Triggering e2e tests

To trigger the E2E test for each new Release added in this PR add a comment with the following:

`/run releases-test-suites`

If you want to trigger conformance tests you can do so by adding a comment similar to the following:

`/run conformance-tests RELEASE_VERSION=v28.0.0 PROVIDER=capa`

For more details see the [README.md](/README.md#running-tests-against-prs)

### CODE REVIEW

File: azure/v25.0.0/release.yaml [hunk 1]
```diff
@@ -109,7 +109,7 @@ spec:
   components:
   - name: cluster-azure
     catalog: cluster
-    version: 0.17.0
+    version: 0.18.0
```
Comment (nprokopic) [inline azure/v25.0.0/release.yaml:112]: Why change this for v25.0.0? We should not modify existing releases.

Comment (njuettner) [reply azure/v25.0.0/release.yaml:112]: Changed the wrong file 🤦🏻

File: azure/v26.0.0/release.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  - name: cluster-azure
+    catalog: cluster
+    version: 0.18.0
+  - name: flatcar
+    version: 3815.2.4
+  - name: flatcar-variant
+    version: 1.0.0
+  - name: kubernetes
+    version: 1.26.15
+  date: "2024-06-27T18:00:00Z"
```
Comment (njuettner) [inline azure/v26.0.0/release.yaml:119]: Just a note, we need to update the date once we merge it

**Comment by njuettner** [comment]
Conformance Tests are the usual ones, which are expected to fail.

File: azure/v26.0.0/release.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
+apiVersion: release.giantswarm.io/v1alpha1
+kind: Release
+metadata:
+  name: azure-26.0.0
+spec:
+  apps:
+  - name: azuredisk-csi-driver
```
Comment (njuettner) [inline azure/v26.0.0/release.yaml:7]: @giantswarm/team-phoenix shall we update azuredisk-csi-driver it seems a bit outdated but I don't know what you plan is here.
We could easily upadate to [1.30.x](https://github.com/kubernetes-sigs/azuredisk-csi-driver?tab=readme-ov-file#container-images--kubernetes-compatibility)

Comment (njuettner) [reply azure/v26.0.0/release.yaml:7]: Opened a PR, https://github.com/giantswarm/azuredisk-csi-driver-app/pull/72

**Review by Gacko — APPROVED** [review]
Apart from the merge conflict remains (please remove them): LGTM!

**Review by yulianedyalkova — APPROVED** [review]

**Pull request merged**